packages/admin/admin/src/form/FinalFormFileUpload.tsx

```diff
@@ -0,0 +1,222 @@
+            cursor: "pointer",
+            minWidth: "250px",
+            "&$disabled": {
+                cursor: "default",
+            },
+            "&$error": {
+                border: `1px dashed ${palette.error.main}`,
+            },
+        },
+        droppableAreaCaption: {
+            color: palette.text.secondary,
+            fontSize: "14px",
+            fontStyle: "normal",
+            fontWeight: "300",
+            lineHeight: "20px",
+        },
+        droppableAreaError: {
+            float: "right",
+            position: "absolute",
+            top: 0,
+            right: 0,
+            padding: "10px",
+            color: palette.error.main,
+        },
+        fileList: {
+            display: "flex",
+            flexDirection: "column",
+            alignItems: "flex-start",
+            gap: "2px",
+            alignSelf: "stretch",
+        },
+        fileListItem: {
+            display: "flex",
+            height: "35px",
+            padding: "8px 7px 8px 15px",
+            alignItems: "center",
+            gap: "10px",
+            alignSelf: "stretch",
+            borderRadius: "4px",
+            background: palette.background.default,
+            justifyContent: "space-between",
+            minWidth: "250px",
+        },
+        rejectedFileListItem: {
+            display: "flex",
+            height: "35px",
+            padding: "8px 7px 8px 15px",
+            alignItems: "center",
+            gap: "10px",
+            alignSelf: "stretch",
+            borderRadius: "4px",
+            background: palette.background.default,
+            justifyContent: "space-between",
+            minWidth: "250px",
+            border: `1px dashed ${palette.error.main}`,
+            color: palette.error.main,
+        },
+        errorMessage: {
+            display: "flex",
+            alignItems: "center",
+            alignSelf: "stretch",
+            color: palette.error.main,
+            gap: "5px",
+        },
+        disabled: {},
+        error: {},
+    });
+};
+
+export interface FinalFormFileUploadProps extends FieldRenderProps<File[], HTMLInputElement> {
+    dropzoneVariant: "dropzoneOnly" | "buttonOnly" | "default";
+    accept: Accept;
+    maxSize: number;
+}
+
+const FinalFormFileUploadComponent: React.FunctionComponent<WithStyles<typeof styles> & FinalFormFileUploadProps> = ({
+    classes,
+    disabled,
+    dropzoneVariant,
+    accept,
+    maxSize = 50,
+    input: { onChange, value: fieldValue, multiple: multipleFiles },
+}) => {
+    const onDrop = React.useCallback(
+        (acceptedFiles: File[]) => {
+            multipleFiles ? onChange([...fieldValue, ...acceptedFiles]) : onChange([...acceptedFiles]);
+        },
+        [fieldValue, multipleFiles, onChange],
+    );
+
+    const removeFile = (file: File) => () => {
+        const newFiles = [...fieldValue];
+        newFiles.splice(newFiles.indexOf(file), 1);
```

Comment:
nit: Is there a reason you use `splice` and `indexOf` instead of `filter` here? I personally think `filter` is easier understandable